android/app/src/main/kotlin/com/yubico/authenticator/piv/PivManager.kt

```diff
@@ -0,0 +1,967 @@
+            }
+        }
+
+    private suspend fun changePin(pin: CharArray, newPin: CharArray): String =
+        connectionHelper.useSmartCardConnection(::updatePivState) {
+            try {
+                val piv = getPivSession(it)
+                handlePinPukErrors { PivmanUtils.pivmanChangePin(piv, pin, newPin) }
+            } finally {
+                Arrays.fill(newPin, 0.toChar())
+                Arrays.fill(pin, 0.toChar())
+            }
+        }
+
+    private suspend fun changePuk(puk: CharArray, newPuk: CharArray): String =
+        connectionHelper.useSmartCardConnection(::updatePivState) {
+            try {
+                val piv = getPivSession(it)
+                handlePinPukErrors { piv.changePuk(puk, newPuk) }
+            } finally {
+                Arrays.fill(newPuk, 0.toChar())
+                Arrays.fill(puk, 0.toChar())
+            }
+        }
+
+    private suspend fun setManagementKey(
+        managementKey: ByteArray,
+        keyType: ManagementKeyType,
+        storeKey: Boolean
+    ): String =
+        connectionHelper.useSmartCardConnection(::updatePivState) {
+            val piv = getPivSession(it)
+            doVerifyPin(piv, pivViewModel.currentSerial())
+            doAuthenticate(piv, pivViewModel.currentSerial())
+            PivmanUtils.pivmanSetMgmKey(
+                piv,
+                newKey = managementKey,
+                algorithm = keyType,
+                touch = false,
+                storeOnDevice = storeKey
+            )
+            ""
+        }
+
+    private suspend fun unblockPin(puk: CharArray, newPin: CharArray): String =
+        connectionHelper.useSmartCardConnection(::updatePivState) {
+            try {
+                val piv = getPivSession(it)
+                handlePinPukErrors { piv.unblockPin(puk, newPin) }
+            } finally {
+                Arrays.fill(newPin, 0.toChar())
+                Arrays.fill(puk, 0.toChar())
+            }
+        }
+
+    private fun handlePinPukErrors(block: () -> Unit): String {
+        try {
+            block()
+            return JSONObject(mapOf("status" to "success")).toString()
+        } catch (invalidPin: InvalidPinException) {
+            return JSONObject(
+                mapOf(
+                    "status" to "invalid-pin",
+                    "attemptsRemaining" to invalidPin.attemptsRemaining
+                )
+            ).toString()
+        } catch (apduException: ApduException) {
+            if (apduException.sw == SW.CONDITIONS_NOT_SATISFIED) {
+                return JSONObject(mapOf("status" to "pin-complexity")).toString()
+            }
+        }
+        return JSONObject(mapOf("status" to "other-error")).toString()
+    }
+
+    private fun getSlots(piv: YubiKitPivSession): List<PivSlot> =
+        try {
+            Slot.entries.minus(Slot.ATTESTATION).map {
+                getSlot(it, piv)
+            }
+        } finally {
+
+        }
+
```

Comment:
There are a few of these empty finally-blocks in this file. Do they actually do anything?